### PR TITLE
Handle keywords in macro fragments

### LINKED ISSUE #2192: Parse ```self``` as ```$_:ident```
I tried this code:

```rust
macro_rules! foo { ($_:ident) => {} }

pub fn foo() {
    foo!(self);
}
```

I expected to see this happen: no error

Instead, this happened:

```
test.rs:4:5: error: Failed to match any rule within macro
    1 | macro_rules! foo { ($_:ident) => {} }
      | ~~~~~~~~~~~
......
    4 |     foo!(self);
      |     ^~~
```

### Meta

git: ```bb1e40da864addfc2d8dfc310928b7a4e7b13185```

### PR DESCRIPTION
**Pull request by powerboat9**
Fixes #2192

### CODE REVIEW

File: gcc/rust/parse/rust-parse-impl.h [hunk 1]
```diff
@@ -999,6 +999,89 @@ Parser<ManagedTokenSource>::parse_delim_token_tree ()
     }
 }
 
+// checks if a token is an identifier/keyword
+template <typename ManagedTokenSource>
+bool
+Parser<ManagedTokenSource>::is_ident_like_tk (TokenId id)
```
Comment (philberty) [inline gcc/rust/parse/rust-parse-impl.h:1005]: I think is_keyword_token would be a better name

**Review by philberty — CHANGES_REQUESTED** [review]
This looks good to me just a small nit with the naming of that helper. Other wise great job!

**Comment by powerboat9** [comment]
Renamed to ```is_keyword_or_ident_token```since it also matches identifiers

**Review by CohenArthur — COMMENTED** [review]
Looks good to me, just wondering about the public parsing function you introduced - we can name it `parse_ident_token` instead and it would be clearer IMO.

**Review by CohenArthur — APPROVED** [review]

**Pull request merged**